src/Fable.Cli/Entry.fs

```diff
@@ -228,19 +228,14 @@ type Runner =
             "FABLE_COMPILER"
             "FABLE_COMPILER_3_OR_GREATER"
             "FABLE_COMPILER_4"
+            "FABLE_COMPILER_4_OR_GREATER"
```

Comment:
Hmm, I was thinking we should just use `FABLE_COMPILER_4_OR_GREATER` and remove `FABLE_COMPILER_4`, do you think it's better to use both? If we use FABLE_COMPILER_4_OR_GREATER only at the end we should also change the other instances below.

Reply:
Personally, I would prefer `FABLE_COMPILER_3` and `FABLE_COMPILER_4` as we don't know if the code we are writing will really be supported with future version of Fable. 

We only know that we support the currently existing version of Fable.

Reply:
I'm agnostic to this so that's why I added both options similar to .NET. But I will let you decide if I should remove one: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation

Reply:
I didn't know that you where basing your proposition on how .NET does it.

For me `FABLE_COMPILER_3` and `FABLE_COMPILER_4` seems enough from what I saw in Haxe worlds. 

But I understand if we want to stick to what .NET does if we feel it to be a good solution too.

Reply:
Then I think we should probably stick with both `FABLE_COMPILER_4` and `FABLE_COMPILER_4_OR_GREATER`.  What do you think @alfonsogarciacaro Should we merge this as is?

Reply:
Ah, you're right @dbrattli! I thought .NET only used the X_OR_GREATER pattern 😅 I'd rather use only the X_OR_GREATER pattern for simplicity but it's true we likely should use both for consistency. In principle we should avoid breaking changes in future versions but we never know. As far as I know the only library using `FABLE_COMPILER_3` is Fable.SimpleJson but this is because before it was relying on the internal JS representation of some types like Map, and this can indeed change in future versions (even minor versions) so it's not a good practice.

Reply:
Ok, great. Let's try this. Fable.SimpleJson is an obvious candidate since we now have Fable.SimpleJson.Python. Fable.JsonProvider also btw have a single line that needs to be handled differently with different languages. I'll merge this PR now. Thanks for reviewing!!